helm/cluster-azure/values.schema.json

```diff
@@ -385,6 +425,44 @@
                     "title": "Location",
                     "type": "string"
                 },
+                "network": {
+                    "additionalProperties": false,
+                    "description": "Azure VNet peering and other Azure-specific network settings.",
+                    "properties": {
+                        "peerings": {
+                            "default": [],
+                            "description": "Specifying VNets (their resource groups and names) to which the peering is established.",
+                            "items": {
+                                "additionalProperties": false,
+                                "properties": {
+                                    "remoteVnetName": {
```

Comment:
```
    providerSpecific:
      network:
        peerings:
        - resourceGroup: glippy
          remoteVnetName: glippy-vnet
```

should we really expect customers to specify this values for each WC cluster ? 

I wonder if we could get those values to automatically generate from the `MC cluster` they get applied to

Reply:
> i think the managementCluster: XXXX is part of every app values by default ( handled by the app operator )

I wasn't aware before that we have `managementCluster: XXXX` available in config, only recently discovered that we have that `cluster-catalog` ConfigMap.

Even with that we would have to introduce MC vs. WC logic here in the `cluster-azure` app because we have this peering only for WCs, and I think we generally don't want to do that, at least didn't want for other `cluster-xxx` apps.

Another totally different way is to have these peerings in the MC's AzureCluster CR, but we woul dhave to add them there dynamically, probably via an operator that watches `AzureCluster` CR creation.

I would maybe even move this to internal, as it could be subject to change when we change how we do VPN.

Reply:
> Even with that we would have to introduce MC vs. WC logic here in the cluster-azure app because we have this peering only for WCs, and I think we generally don't want to do that, at least didn't want for other cluster-xxx apps.

That being said, I'll try something and we see how it looks like. Might be fine for the sake of nicer UX, especially if this is just a short-term solution.

Reply:
I tried to make this WC to MC peering implicit in WCs. Templates worked locally, can't get it to work when deploying the cluster.

Reply:
VNet peering from WC to MC VNets is now added automatically for private clusters, the explicit config is not required anymore.